includes/object-cache.php

```diff
@@ -1629,6 +1642,75 @@ public function flush( $delay = 0 ) {
         return true;
     }
 
+    /**
+	 * Removes all cache items in a group.
+	 *
+	 * @param string $group Name of group to remove from cache.
+	 * @return true Returns TRUE on success or FALSE on failure.
+	 */
+    public function flush_group( $group )
+    {
+        $san_group = $this->sanitize_key_part( $group );
+
+        if ( is_multisite() && ! $this->is_global_group( $san_group ) ) {
+            $salt = str_replace( "{$this->blog_prefix}:", '*:', $this->fast_build_key( '*', $san_group ) );
+        } else {
+            $salt = $this->fast_build_key( '*', $san_group );
+        }
+
+        $this->cache = []; // TODO: remove all matching keys from  $this->cache
```

Comment:
This is looking good, just needs to iterate over `$this->cache` and delete all keys that match the pattern. Because this is internal we can just use:

```php
strpos($id, "{$san_group}:") === 0 || strpos($id, ":{$san_group}:") !== false
```